Attribute Analysis.ipynb

```diff
@@ -0,0 +1,650 @@
+    "        plt.barh(r[i], scenes_ratio[i][order], height=barWidth, color=COLORS[i], edgecolor='white', label=attr_names[i])\n",
+    "    ticks = r[0]+(num_attrs/2)*barWidth\n",
+    "    plt.yticks(ticks, np.array(xaxis)[order], fontsize=fontsize)\n",
+    "    plt.xticks(fontsize=fontsize)\n",
+    "    plt.ylabel('Scene', fontsize=fontsize)\n",
+    "    #May want to add attribute name in graph title\n",
+    "    plt.xlabel('Proportion of Images Discarded with this Scene', fontsize=fontsize)\n",
+    "    plt.legend(loc='best', prop={'size': fontsize})\n",
+    "    plt.title(\"Scenes where image was labeled when it should not have been\", fontsize=fontsize)\n",
+    "    plt.tight_layout()\n",
+    "    plt.gcf().subplots_adjust(bottom=0.18)\n",
+    "    plt.gcf().subplots_adjust(left=0.4)\n",
+    "    plt.show()\n",
+    "    \n",
+    "comparisons_widget = widgets.Dropdown(options=['Sizes', 'Distances', 'All sizes', 'Sizes where no face was detected'], value='Sizes')\n",
+    "\n",
+    "def compare_sizedist(metric, view):\n",
+    "    def mean_and_std(data, data_type):\n",
+    "        mean = [np.mean(data[i]) for i in range(num_attrs)]\n",
+    "        std = [np.std(data[i]) for i in range(num_attrs)]\n",
+    "\n",
+    "        #May want to change, atm it takes the least significant p, and if that's less than 0.05 then proceeds\n",
+    "        min_p = 100\n",
+    "        one = -1\n",
+    "        two = -1\n",
+    "        for i in range(num_attrs):\n",
+    "            for a in range(i+1, num_attrs):\n",
+    "                t, p = stats.ttest_ind(data[i], data[a])\n",
+    "                if p < min_p:\n",
+    "                    min_p=p\n",
+    "                    one = i\n",
+    "                    two = a\n",
+    "        p = min_p\n",
+    "\n",
+    "        #May want to specify what \"attributes\" is\n",
+    "        to_save = False\n",
+    "        if view is None and p < .05 and first_pass:\n",
+    "            data_descrip = ''\n",
+    "            if data_type == 'dists':\n",
+    "                data_descrip = 'Distance from center'\n",
+    "            if data_type == 'sizes':\n",
+    "                data_descrip = 'Fraction of image taken up by a person'\n",
+    "            to_write[2] = [\"(M1) {0} is different between the attributes with a p-value of {1} for the most significant pair ({2} and {3}), distribution shown below\".format(data_descrip, dec_to_show(p), attr_names[one], attr_names[two])]\n",
+    "            to_save = True\n",
+    "            \n",
+    "        if view is None or view:\n",
+    "            for i in range(num_attrs):\n",
+    "                histogram_a, bins_a = np.histogram(data[i], bins='auto')\n",
+    "                bin_centers_a = 0.5*(bins_a[1:] + bins_a[:-1])\n",
+    "                area_a = np.trapz(histogram_a, x=bin_centers_a)\n",
+    "                #Add more specific graph titles that feature attribute name?\n",
```

Comment:
if these comments are for leslie when she takes a pass, please make a note elsewhere and send to her, but don't include them as comments in the code